js/web/lib/wasm/jsep/webgpu/gpu-data-manager.ts

```diff
@@ -76,9 +81,12 @@ class GpuDataManagerImpl implements GpuDataManager {
   // pending buffers for computing
   private buffersPending: GPUBuffer[];
 
+  private freeBuffers: Map<number, GPUBuffer[]>;
+
   constructor(private backend: WebGpuBackend /* , private reuseBuffer: boolean */) {
     this.storageCache = new Map();
     this.downloadCache = new Map();
+    this.freeBuffers = new Map();
```

Comment:
Are all buffers stored in this map always in `unmapped` state?

Reply:
Yes. Currently all buffers in this map are storage buffers with usage `GPUBufferUsage.STORAGE | GPUBufferUsage.COPY_SRC | GPUBufferUsage.COPY_DST` and don't need map/unmap for such kind of buffers. The upload buffers and download buffers are not in this map.
Maybe we can also reuse the download buffers in future. But I prefer we use different maps. Otherwise, the key needs to be usage + bufferSize.